src/MediaPresenter.ts

```diff
@@ -0,0 +1,139 @@
+import * as mobx from "mobx";
+
+const MAXLEN = 10;
+
+export class MediaPresenter {
+  constructor() {
+    mobx.makeObservable(this);
+  }
+  // make need to change to .deep
+  @mobx.observable.deep
+  files: File[] = [];
+
+  @mobx.observable.deep
+  durations: number[] = [];
+  // for now its a 1D array but can be expanded as [][] to play different parts
+  // this is for videos, which parts of it have been played
+  played: number[] = [];
+
+  customOrder: boolean = false;
+
+  static audioSound: number = 3;
+
+  static isImage(file: File): boolean {
+    const imageFormat = new RegExp("image/*");
+    return imageFormat.test(file.type);
+  }
+
+  static isVideo(file: File): boolean {
+    const videoFormat = new RegExp("video/*");
+    return videoFormat.test(file.type);
+  }
+
+  @mobx.action
+  addFile(newFile: File): boolean {
+    if (MediaPresenter.isImage(newFile) || MediaPresenter.isVideo(newFile)) {
+      this.files.push(newFile);
+      this.durations.push(0);
+      this.played.push(0);
+      return true;
+    } 
+    return false;
+  }
+  
+  @mobx.action
+  removeFile(fileIndex: number): void {
+    let index = fileIndex % this.filesLength;
+    if (index > -1) {
+      this.files.splice(index, 1);
+      this.durations.splice(index, 1);
+      this.played.splice(index, 1);
+    }
+  }
+
+  setCustomOrder(value: boolean) {
+    this.customOrder = value;
+  }
+
+  @mobx.action
+  setDuration(fileIndex: number, duration: number): void {
+    this.durations[fileIndex % this.filesLength] = duration;
```

Comment:
create a const for `fileIndex % this.filesLength` with a comment to explain what it's doing, same in the other fns
alternately, make a `computed` value for this calculation